### PR TITLE
Adds IT for RPM artifacts

### PR DESCRIPTION
**Pull request by gaiksaya**
Adds IT for RPM artifacts
Self hosted runner is set up and is actively working. 

I have [tested](https://github.com/gaiksaya/opendistro-build/actions/runs/112947452) in my repo.

### CODE REVIEW

File: .github/workflows/test-build-rpm.yml [hunk 1]
```diff
@@ -191,3 +191,153 @@ jobs:
...
+          echo "::set-env name=tag::$plugin_tag"
+
+      - name: Checking out ISM repo
+        if: always()
+        uses: actions/checkout@v1
+        with:
+          repository: opendistro-for-elasticsearch/index-management
+          ref: ${{env.tag}}
+
+      - name: IT for ISM
```
Comment (weicongs-amazon) [inline .github/workflows/test-build-rpm.yml:261]: just keep in mind that sharing cluster might cause some execution issue even if it works well for all of the plugins now

Comment (gaiksaya) [reply .github/workflows/test-build-rpm.yml:261]: Yeah I had that thought! One solution might be clearing up after each test and then installing/starting service again for next plugin
Another solution might be using different self hosted runners for all plugins

File: .github/workflows/test-build-rpm.yml [hunk 2]
```diff
@@ -191,3 +191,153 @@ jobs:
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Configuring AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_EC2_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_EC2_SECRET_ACCESS_KEY }}
+          aws-region: us-west-2
+
+      - name: Resetting the host runner
```
[the remark this answers is not in the document]
Comment (gaiksaya) [reply .github/workflows/test-build-rpm.yml:322]: manual intervention would be required for now. We will monitor and add additional steps accordingly or I can add a step to create a new ec2 with same config

[the remark this answers is not in the document]
Comment (weicongs-amazon) [reply .github/workflows/test-build-rpm.yml:322]: let's monitor this for a while firstly. just clearly post the manual instructions somewhere, hence any team member can handle this issue

[the remark this answers is not in the document]
Comment (gaiksaya) [reply .github/workflows/test-build-rpm.yml:322]: Sure will do!

**Pull request merged**